### PR TITLE
highlight white spaces in diffs

### PR DESCRIPTION
**Pull request by premek**
when there was a difference only in whitespace it wasn't highlighted in the diff

Screenshot from 2023-03-24 18-09-14

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -227,7 +227,7 @@
           const isLoading = Files.isLoading(vnode.attrs.path);
           return text === "" ? m('p', m("i.gray", isLoading ? "Loading" : "Empty file")) : m("pre.f6", {
             style: "white-space: pre-wrap; tab-size: 4;",
-          }, text ? text.split("\n").map(l => [l, m('span', {title: "newline", style: "user-select: none; color: #eaeaea;"}, "↵"), "\n"]) : "");
+          }, text ? text.split("\n").flatMap(l => [m('span', {title: "newline", class: "moon-gray", style: "user-select: none"}, "↵"), "\n", l]).slice(2) : "");
```
Comment (premek) [inline index.html:230]: this was adding an extra newline symbol at the end

Comment (premek) [reply index.html:230]: I changed the font color of the newline symbol to just a slightly more visible one :)

File: index.html [hunk 2]
```diff
@@ -245,8 +245,8 @@
             return m("pre.f6", {
               style: "white-space: pre-wrap; tab-size: 4;",
             }, diff.map(p => {
-                const c = p.added ? 'green' : p.removed ? 'red' : '';
-                return m('span', {class: c}, p.value);
+                const c = p.added ? 'bg-washed-green dark-green' : p.removed ? 'bg-washed-red dark-red' : '';
+                return p.value.split("\n").flatMap(l => [m('span', {title: "newline", class: c || "moon-gray", style: "user-select: none"}, "↵"), "\n", m('span', {class: c}, l)]).slice(2);
```
Comment (premek) [inline index.html:249]: - added red / green background (useful when the text is just a space for example)
- added newline symbols to the diff

**Comment by chromy** [comment]
Thanks! Much nicer 😃

**Pull request merged**